### ISSUE TITLE
Make twisted/python/logfile.py Python3 syntax compatible

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7142]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#7142|
|Type|enhancement|
|Created|2014-04-14 17:45:52Z|
|Branch|https://github.com/twisted/twisted/tree/logfile-py3-7142|

Right now it's not. This blocks trial from supporting Python3.

Attachments:

* [t7142.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e3d/e3d595b0d25d8b012c23a9dde58c2bd8f1c71272/e04cfab4329b20940850e6744809bf81ca1b259b.diff) (1215 bytes) - added by Alex on 2014-04-14 17:46:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7142 7142
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_logfile_py3_7142 branches/logfile-py3-7142
branch_author__radix radix
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1397497552000000 1397497552000000
changetime__1437250832420694 1437250832420694
version__None None
owner__Alex Alex

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix commented|
|-|-|

(In [42208]) Branching to logfile-py3-7142.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix set owner to Alex|
|-|-|

This is a good change. Thanks. If the buildbots pass you can merge it.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix set status to `closed`|
|-|-|

(In [42211]) Merge logfile-py3-7142: python 3 syntax for logfile.py

Author: Alex
Reviewer: radix
Fixes: [#7142](https://github.com/twisted/twisted/issues/7142)

Update syntax of twisted/python/logfile.py to be compatible
with Python 3.

**Comment by twisted-trac** [comment]
|<img alt="jMyles's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|jMyles commented|
|-|-|

It's not listed in dist3, though.  Is it supposed to be?

Adding it does not cause any test failures, but it's not clear to me which, if any, modules in twisted.python.test are meant to be added to dist3 alongside twisted.python.logfile.
